### PR TITLE
*: update tidb dependency build with go1.14

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
update tidb(https://github.com/pingcap/tidb/pull/14972) to make go build success in go1.14
also need fix this https://github.com/pingcap/tidb/issues/15093 first
### What is changed and how it works?
add `AllocSeqCache` function in `idAlloctor` to adapt new tidb `Allocator` interface

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: pkg/restore/util.go [hunk 1]
```diff
@@ -36,6 +36,10 @@ func (alloc *idAllocator) Alloc(tableID int64, n uint64, increment, offset int64
 	return alloc.id, alloc.id, nil
 }
 
+func (alloc *idAllocator) AllocSeqCache(sequenceID int64) (min int64, max int64, round int64, err error) {
+	return 0, 0, 0, nil
```
Comment (overvenus) [inline pkg/restore/util.go]: Is it safe to return zero if we restore Sequence?

Comment (3pointer) [reply pkg/restore/util.go]: do we support backup sequence?

Comment (kennytm) [reply pkg/restore/util.go]: i don't think backup sequence need to go through this function

**Comment by kennytm** [comment]
Is this PR still WIP?

**Comment by 3pointer** [comment]
> Is this PR still WIP?

yes, need fix https://github.com/pingcap/tidb/issues/15093 first

**Pull request merged**